File: patient/src/main/java/com/openspace/PatientManagement/service/PaymethodService.java
package com.openspace.PatientManagement.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.openspace.Model.DoctorManagement.Paymethod;

public interface PaymethodService {

	void addPaymethod(Paymethod paymethod);

	List<Paymethod> getAllPaymethods();

	void updatePaymethod(Paymethod paymethod);

	void deletePaymethod(Long id);

	List<Paymethod> getAllPaymethodsByAdmin(String username);

	Page<Paymethod> getAllPaymethodsByAdminPage(String username, int page, int size);
}

File: portal/src/main/resources/static/service/finance/paymethodService.js
function paymethodService(PATIENT_MODULE_CONFIG,MODULE_CONFIG, $http) {
	this.addPaymethod = function(paymetod) {
	return 	$http.post(PATIENT_MODULE_CONFIG.ADD_PAYMETHOD(),paymetod);
	};
	this.getAllPaymethods=function(email){
		return $http.get(PATIENT_MODULE_CONFIG.GETALLPAYMETHODS(email));
	};
	this.getAllPaymethodsList=function(email,page, size){
		return $http.get(PATIENT_MODULE_CONFIG.GET_ALL_PAYMETHODS_LIST(email,page, size));
	};
	this.updatePaymethod=function(paymetod){
		return $http.put(PATIENT_MODULE_CONFIG.UPDATE_PAYMETHOD(),paymetod);
	};
	this.deletePaymethod=function(id){
		return $http.delete(PATIENT_MODULE_CONFIG.DELETE_PAYMETHOD(id));
	};
	
}
angular.module('HealthApplication').service('paymethodService', paymethodService);